services/motion/builtin/move_request.go

```diff
@@ -193,8 +191,35 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 		return nil, err
 	}
 
+	kinematicFrame := kb.Kinematics()
 	// replace original base frame with one that knows how to move itself and allow planning for
-	if err = fs.ReplaceFrame(kb.Kinematics()); err != nil {
+	if err = fs.ReplaceFrame(kinematicFrame); err != nil {
+		return nil, err
+	}
+	// We want to disregard anything in the FS whose eventual parent is not the base, because we don't know where it is.
+	baseOnlyFS, err := fs.FrameSystemSubset(kinematicFrame)
+	if err != nil {
+		return nil, err
+	}
+	// Place the base at the correct heading relative to the frame system
+	currentPosition, err := kb.CurrentPosition(ctx)
+	if err != nil {
+		return nil, err
+	}
+	headingFrame, err := referenceframe.NewStaticFrame(
+		kinematicFrame.Name()+"_heading",
```

Comment:
[q]: how is the `_heading` suffix understood by the frame system / motion planner?
I know that there is also the `_origin` suffix, which I also do not understand too well.. Where in the codebase can I learn more about them?

Reply:
But to answer your question, the motion planner doesn't care about "_origin" frames specially at all; the only time they are treated specially is at creation, here:
https://github.com/viamrobotics/rdk/blob/700cbba84795c3f4480eec52e3ea25ecb81ef369/referenceframe/frame_system.go#L615